core/process.py

```diff
@@ -399,10 +408,21 @@ def _ApplyRedirect(self, r):
                 try:
                     open_fd = posix.open(arg.filename, mode, 0o666)
                 except (IOError, OSError) as e:
-                    self.errfmt.Print_("Can't open %r: %s" %
-                                       (arg.filename, pyutil.strerror(e)),
-                                       blame_loc=r.op_loc)
-                    raise  # redirect failed
+                    if noclobber != 0 and e.errno == EEXIST:
+                        self.errfmt.PrintMessageEx(
+                            "I/O redirect error: can't overwrite existing file %r: %s"
+                            % (arg.filename, pyutil.strerror(e)),
+                            self.mutable_opts.Get(option_i.verbose_errexit),
+                            r.op_loc)
+                    else:
+                        self.errfmt.PrintMessageEx(
+                            "I/O redirect error: can't open file %r: %s" %
+                            (arg.filename, pyutil.strerror(e)),
+                            self.mutable_opts.Get(option_i.verbose_errexit),
```

Comment:
Hm why not just use PrintMessage() ?  It already doesn't print the code?

I agree that there were some duplicate messages from redirects.  It's a little hard to sort out, because there are multiple cases.    Sometimes it prints too much, sometimes too little, etc.


   `verbose_errexit`  isn't really the right option here anyway -- that is for `set -e` failures

Reply:
That's right, this is completly wrong...